src/recorder/jamcontroller.cpp

```diff
@@ -77,15 +78,25 @@ void CJamController::SetRecordingDir ( QString newRecordingDir,
                                        int     iServerFrameSizeSamples,
                                        bool    bDisableRecording )
 {
-    if ( bRecorderInitialised && pthJamRecorder != nullptr )
+    if ( bRecorderInitialised )
     {
-        // We have a thread and we want to start a new one.
-        // We only want one running.
-        // This could take time, unfortunately.
-        // Hopefully changing recording directory will NOT happen during a long jam...
-        emit EndRecorderThread();
-        pthJamRecorder->wait();
-        pthJamRecorder = nullptr;
+        if ( pthJamRecorder != nullptr )
+        {
+            // We have a thread and we want to start a new one.
+            // We only want one running.
+            // This could take time, unfortunately.
+            // Hopefully changing recording directory will NOT happen during a long jam...
+            emit EndRecorderThread();
+            pthJamRecorder->wait();
+            delete pthJamRecorder;
+            pthJamRecorder = nullptr;
+        }
+        if ( pJamRecorder != nullptr )
```

Comment:
Not sure about this new bit.  Can you explain why you lose the JamRecorder main thread reference?

Reply:
The old code discarded the value of `pthJamRecorder` without freeing what it was pointing to. That was one of the leaks. Nowhere else was pointing to it, as far as I could tell.

Reply:
pthJamRecorder and pJamRecorder are different things.  You've added code for pJamRecorder.  That's what I'm questioning.  I understand the pthJamRecorder (thread-specific) code.

Reply:
That's because a couple of lines further down there is:
```
pJamRecorder = new recorder::CJamRecorder ( newRecordingDir, iServerFrameSizeSamples );
```
which will leak whatever `pJamRecorder` was pointing to before. So my addition cleans it up before it gets overwritten.

Reply:
But that won't always happen - `if (!newRecordingDir.isEmpty())`.  Any management of `pJamRecorder` should be just before creating a new one.

Reply:
See comment below...